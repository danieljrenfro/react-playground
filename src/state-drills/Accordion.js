import React, { Component } from 'react';

class Accordion extends Component {
  state = {
    currentSection: null
  }

  static defaultProps = {
    sections: []
  }

  onClick = (index) => {
    this.setState({ currentSection: index });
  }
  
  render() {
    const buttons = this.props.sections.map((section, i) => {
      return <li key={i}>
        <button onClick={() => this.onClick(i)} type="button">{section.title}</button>
        {this.state.currentSection === i && <p>{section.content}</p>}
      </li>
    })

    return (
      <ul>
        {buttons}
      </ul>
    )
  }
}

export default Accordion;