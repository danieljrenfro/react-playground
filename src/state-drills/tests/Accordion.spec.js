import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Accordion from '../Accordion';

import sections from '../accordion-sections';

describe('Accordion component', () => {
  it('renders without crashing without props', () => {
    const div = document.createElement('div');
    
    ReactDOM.render(<Accordion />, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders without crashing with props', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<Accordion sections={sections} />, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly given section', () => {
    const wrapper = shallow(<Accordion sections={sections}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens the first section when button clicked', () => {
    const wrapper = shallow(<Accordion sections={sections}/>);
    wrapper.find('button').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('only opens the last section when it is clicked, not before', () => {
    const wrapper = shallow(<Accordion sections={sections}/>);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})