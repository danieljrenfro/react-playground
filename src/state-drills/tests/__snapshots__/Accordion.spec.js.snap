// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component only opens the last section when it is clicked, not before 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion component opens the first section when button clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders correctly given section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders empty given no sections 1`] = `<ul />`;
